src/main/java/codeu/model/data/Activity.java

```diff
@@ -0,0 +1,112 @@
+package codeu.model.data;
+
+import codeu.model.util.Util;
+
+import java.time.Instant;
+import java.util.UUID;
+
+/**
+ * Class representing an activity (creating user, message, or conversation). Any creation done by a
+ * user prompts the creation of an activity object.
+ */
+public class Activity {
+  private final UUID id;
+  private final UUID ownerId;
+  private final Action action;
+  private Boolean isPublic;
```

Comment:
I think this should be a primitive `boolean` unless you're specifically allowing for the case where the activity isn't specified as either public or non-public (i.e., `isPublic == null`).

Reply:
I don't think isPublic should equal null so I will change it. Maybe add it in the future if we encountered such activity